src/shared/utilities/promptSetting.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { getLogger } from '../logger'
+import { SettingsConfiguration } from '../settingsConfiguration'
+
+export type PROMPT_NAME =
+    | 'warnBeforeRunCommand'
+    | 'warnBeforeEnablingExcecuteCommand'
+    | 'warnBeforeDisablingExecuteCommand'
+export const DEFAULT_PROMPTS: PromptSetting = {
+    warnBeforeRunCommand: true,
+    warnBeforeEnablingExcecuteCommand: true,
+    warnBeforeDisablingExecuteCommand: true,
+}
+
+export interface PromptSetting {
+    [promptName: string]: boolean
+}
+
+export function disablePrompt(promptName: PROMPT_NAME, settings: SettingsConfiguration): void {
+    try {
+        const prompts = settings.readSetting<PromptSetting>('togglePrompts', DEFAULT_PROMPTS)
+
+        prompts[promptName] = false
```

Comment:
- vscode settings cannot be trusted to return the type given by the type parameter. The user's settings.json could have any JSON value (boolean, string, array, ...). Rather than try/catch we can test for this to show a more informative message.
    - Because we own this setting (it's an "AWS" setting), we can just overwrite a bad value.
- writeSetting() already does try/catch internally, so the try/catch in this scope is unnecessary, if we check the type.

```suggestion
        if (type prompts !== 'object' || Array.isArray(prompts)) {
            getLogger().warn('setting "togglePrompts" has an unexpected type. Overwriting.')
            prompts = {}
        }
        prompts[promptName] = false
```

Reply:
I wonder if the config contribution point can accept a schema to validate individual settings? We should still validate within the code too but both would be ideal.

Reply:
This was very useful, thank you.